### PR TITLE
Issue 342: Custom De-Serialization Bug

### PR DESCRIPTION
**Pull request by rmartinc**
Fix for issue #342 in master branch. Just calling respective methods to do the getObject and getArray. Added a test for parsing an object.
The fix is also needed for 1.x branch if you want another PR please let me know.

### CODE REVIEW

File: src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java [hunk 1]
```diff
@@ -271,7 +271,15 @@ public JsonObject getObject() {
 
     @Override
     public JsonValue getValue() {
-        return jsonParser.getValue();
+        final LevelContext currentLevel = level.peek();
+        switch (currentLevel.getLastEvent()) {
```
Comment (jamezp) [inline src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java:275]: Just a minor note we could use `switch (getLastEvent())`. Either way there is a potential for an NPE here if the `level` queue is empty.

Comment (rmartinc) [reply src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java:275]: I just did the same it is in `skipJsonStructure` method. If you want I can check `isEmpty` first and just delegate to `jsonParser.getValue()` in that case.

Comment (jamezp) [reply src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java:275]: +1 that works too.

Comment (rmartinc) [reply src/main/java/org/eclipse/yasson/internal/JsonbRiParser.java]: It's done! Checking empty and possible null value in the event itself.

**Comment by Verdent** [comment]
Hi @rmartinc ,
Thank you very much for taking time to fix one of our bugs! It would be great if you could create also PR to 1.x branch since this is a bug fix if this is not too much trouble. :-)

**Review by Verdent — APPROVED** [review]
LGTM

**Comment by rmartinc** [comment]
> Hi @rmartinc ,
> Thank you very much for taking time to fix one of our bugs! It would be great if you could create also PR to 1.x branch since this is a bug fix, if this is not too much trouble. :-)

Yes, of course. It's https://github.com/eclipse-ee4j/yasson/pull/482. Thanks to you!

**Pull request merged**